src/os/ui/data/layercheckbox.js

```diff
@@ -68,16 +69,45 @@ class Controller extends TriStateCheckboxCtrl {
         const state = item.getState();
 
         if (state == TriState.OFF) {
-          if (children && children.length > this.minCount) {
+          let count = 0;
+          const countLayerChildren = (child) => {
```

Comment:
We might want to make this a public convenience function on this class (could be static) or elsewhere. That would make it directly testable, and would avoid creating a new function instance every time this function is called. You would just need to add `count` as a parameter/return value to avoid the need for a local closure, then swap  `forEach` with `reduce` to aggregate the count.

Reply:
Yep. I just tend to find local recursive `forEach` more straightforward than `reduce`.

Reply:
I do too, but if a local function closure can be avoided I tend to lean towards that.